lib/services/firebase_service.dart

```diff
@@ -1090,22 +1132,17 @@ class FirebaseService extends IFirebaseService {
   Future<Map<dynamic, dynamic>> getFormDataByLastFormId(
       String formName, String patientState, String hn) async {
     var mapData = {};
-    var formId = await _firestore
+    await _firestore
         .collection('Forms')
         .where('hn', isEqualTo: hn)
         .where('formName', isEqualTo: formName)
         .where('patientState', isEqualTo: patientState)
+        .limit(1)
         .get()
-        .then((value) => value.docs.first.id)
-        .catchError((onError) {});
-
-    if (formId != null) {
-      await _firestore.collection('Forms').doc(formId).get().then((value) {
-        mapData.addAll(value.data());
-      }).catchError((onError) {
-        print('$onError Cann\'t find $formName form');
-      });
-    }
+        .then((value) {
+      mapData.addAll(value.docs.last.data());
+      return mapData;
+    }).catchError((onError) {});
```

Comment:
![image](https://user-images.githubusercontent.com/57885091/117850288-5a5d0a80-b2af-11eb-9d2a-7fcdcb6bcd94.png)

@PormpimonPang  if you want the latest one first `descending = true` since it sort by date. then use `limit(1)` gonna force the query output to be 1. but I don't sure that is it still return as querydocument or documentref.

Reply:
c49cd6b